### PR TITLE
LibWeb + LibJS: Further fixes after the scoping rework

### PR DESCRIPTION
**Pull request by davidot**
I have also created a tests for the unicode issue fixed in #9947.
Most of these fixes are really stop-gap fixes to make things work until we have proper realm/executing context support.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/DeclarativeEnvironment.h [hunk 1]
```diff
@@ -31,6 +31,9 @@ class DeclarativeEnvironment : public Environment {
 
     void initialize_or_set_mutable_binding(Badge<ScopeNode>, GlobalObject& global_object, FlyString const& name, Value value);
 
+    // This is not a method defined in the spec! Do not use this in any LibJS (or other spec related) code.
+    [[nodiscard]] Vector<String> bindings() const;
+
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/DeclarativeEnvironment.h:36]: This feels like it could be dangerous but at the same time adding a `Badge<js>` or something didn't feel like it was much better. We can just avoid this but then we have no autocomplete on:
```
> const abc = 3;
undefined
> a<tab> -> abc
```

File: Tests/LibJS/test-invalid-unicode-js.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/*
```
Comment (davidot) [inline Tests/LibJS/test-invalid-unicode-js.cpp:1]: Not sure if this is how we want to test things like this (and about the name).
But this is the most explicit way I could think of.

**Review by linusg — APPROVED** [review]
Thanks!
